### PR TITLE
Bring repository into REUSE compliance

### PR DESCRIPTION
**Pull request by oddhack**
This now passes REUSE with the following exceptions:

* lib/dfdutils - a new copy should be imported from that repo as the REUSE work has already been done there
* khr_df.h - again a current copy should be imported from the DataFormat repo where REUSE compliance has been done.

There are a vast number of licenses (11, I think) in this repo, due to the many sub-repos and imported files. It was sometimes difficult to recover the entire history and assumptions had to be made. Where possible SPDX identifiers were imbedded directly in files, but in many cases, they are instead captured in .reuse/dep5. I strongly encourage review of the metainformation in dep5, particularly where wildcards are given (the way REUSE works, I believe it is checking the individual files first, only then falling back to the patterns in dep5).

@MarkCallow I don't know how much review you'll feel a need to do here, but I think it would be particularly important to confirm nothing I've done has interfered with configuration and build processes on the Win/Mac/Linux platforms - I've edited a number of files where I'm not completely certain how they're used, though the commenting conventions were pretty obvious.

Going forward, I think the utility of the NOTICE file in the repo is questionable. If we label this repository as multi-license - which IMO it really is, with all these licenses - it seems unlikely that clause 4 in the Apache 2.0 license comes into play, since "The Work" does not constitute the entire repo in that case. ISTM that's the easiest path forward, since the NOTICE file is out of date and incomplete AFAICT.

This repo has such a complex licensing structure that I think it would be a really good idea to add a CI test that runs "reuse lint" as a required pass, going forward and once the dfdutils / khr_df.h issue is dealt with.

### CODE REVIEW

**Comment by oddhack** [comment]
Oh, I also encourage importing new versions of the GL headers, though it's not clear to me if those are bound to the GLEW files also in the tree - if so that might be challenging.

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+License: CC-BY-3.0
+
+Files: tests/srcimages/Iron_Bars/*
+Copyright: Katsuagi https://3dtextures.me
+License: CC0-1.0
+
+# Jon - Unclear whether all remaining images & icons fall under the Khronos license
+Files: */*.png */*.ktx */*.ico */*.icns */*.jpg */*.pam */*.pgm */*.ppm */*.ktx */*.ktx2 */*.basis */*.pspimage
+Copyright: 2015-2020 The Khronos Group Inc.
+License: Apache-2.0
```
Comment (MarkCallow) [inline .reuse/dep5:16]: ccwn2c08.png, g03n2c08.png, color_grid.png and kodim17.png come from outside. The first 2 are from the PNG test suite. The license is at http://www.schaik.com/pngsuite/PngSuite.LICENSE. kodim17.png comes from the Kodak Images repo http://r0k.us/graphics/kodak/.

 I'm struggling to find where color_grid is from. I got it from someone in the 3dformats group. I ping the group to see if I can find out.

camera_camera_BaseColor_16bit.png and FlightHelmet_baseColor.png come from the glTF samples repo](https://github.com/KhronosGroup/glTF-Sample-Models) and are in the public domain.

CesiumLogoFlat_palette.png also comes from the glTF samples repo. The license points to https://github.com/AnalyticalGraphicsInc/cesium/wiki/CesiumTrademark.pdf.

All the  other images & icons fall under the Khronos license.

File: .reuse/dep5 [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+Copyright: 2013 Ericsson AB 2013. All Rights Reserved.
+License: LicenseRef-ETCSLA
+
+# This has an SPDX identifier but REUSE doesn't recognize it
+Files: tests/loadtests/vkloadtests/resources/ios/LaunchScreen.storyboard
+Copyright: 2017-2020 The Khronos Group Inc.
+License: Apache-2.0
+
+# These are resource files generated by Visual C++. Not sure what license
+# they have, if any. Assuming they share the main repo license for now.
```
Comment (MarkCallow) [inline .reuse/dep5:108]: Yes they do. I created them. VC++ was just the tool.

File: .reuse/dep5 [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Files: tests/*/*loadtests.rc
+Copyright: 2020 The Khronos Group Inc.
+License: Apache-2.0
+
+# It's not certain that the stb font file indeed comes from tools in the
+# nothings/stb project on Github, but a reasonable assumption.
```
Comment (MarkCallow) [inline .reuse/dep5:114]: Yes it does.

File: LICENSES/GPL-2.0-or-later.txt [hunk 4]
```diff
@@ -0,0 +1,103 @@
+GNU GENERAL PUBLIC LICENSE
```
Comment (MarkCallow) [inline LICENSES/GPL-2.0-or-later.txt:1]: Why is this  here? It isn't mentioned in .reuse/dep5.

File: install-gitconfig.bat [hunk 5]
```diff
@@ -2,27 +2,9 @@ echo OFF
...
-REM
-REM THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-REM EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-REM MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-REM IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-REM CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-REM TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-REM MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+REM Copyright 2016 The Khronos Group Inc.
+REM SPDX-License-Identifier: MIT
```
Comment (MarkCallow) [inline install-gitconfig.bat:6]: This can be changed to Apache 2.0. I wrote this script and overlooked it when I converted the other Khronos code to Apache 2.0.

File: install-gitconfig.sh [hunk 6]
```diff
@@ -2,27 +2,9 @@
...
-#
-# THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-# MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+# Copyright 2016 The Khronos Group Inc.
+# SPDX-License-Identifier: MIT
```
Comment (MarkCallow) [inline install-gitconfig.sh:6]: Same with this.

File: interface/js_binding/js_binding.gypi [hunk 7]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2015-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2015, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline interface/js_binding/js_binding.gypi:6]: Remove this line.

File: ktxdoc.gyp [hunk 8]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2016-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2016, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline ktxdoc.gyp:6]: Remove this line.

File: ktxtools.gyp [hunk 9]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2016-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2016, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline ktxtools.gyp:6]: Remove this line.

File: lib/basisu/CMakeLists.txt [hunk 10]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2017-2020 The Khronos Group Inc.
+# SPDX-License-Identifier: Apache-2.0
+
 project(basisu)
 
 cmake_minimum_required(VERSION 3.0)
```
Comment (MarkCallow) [inline lib/basisu/CMakeLists.txt:6]: Please make a separate PR for the changes to `lib/basisu`. I need to pull a very large change from Binomial soon and don't want the merge nightmare. I prefer to pull the change then apply a PR for adding SPDX, etc. I might even try to get Binomial to REUSEIFY their repo.

File: lib/libktx.gypi [hunk 11]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2015-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2015, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline lib/libktx.gypi:6]: Remove this line.

File: lib/texture1.c [hunk 12]
```diff
@@ -62,7 +51,7 @@ ktxTexture1_constructCommon(ktxTexture1* This)
     if (This->_private == NULL) {
         return KTX_OUT_OF_MEMORY;
     }
-	memset(This->_private, 0, sizeof(*This->_private));
+        memset(This->_private, 0, sizeof(*This->_private));
```
Comment (MarkCallow) [inline lib/texture1.c:54]: wrong number of spaces in place of the tab. Should only be 4.

File: lib/texture1.c [hunk 13]
```diff
@@ -82,7 +71,7 @@ ktxTexture1_construct(ktxTexture1* This, ktxTextureCreateInfo* createInfo,
     GLenum glFormat;
     KTX_error_code result;
 
-	memset(This, 0, sizeof(*This));
+        memset(This, 0, sizeof(*This));
```
Comment (MarkCallow) [inline lib/texture1.c:74]: Same here.

File: lib/texture1.c [hunk 14]
```diff
@@ -210,7 +199,7 @@ ktxTexture1_constructFromStreamAndHeader(ktxTexture1* This, ktxStream* pStream,
 
     assert(pHeader != NULL && pStream != NULL);
 
-	memset(This, 0, sizeof(*This));
+        memset(This, 0, sizeof(*This));
```
Comment (MarkCallow) [inline lib/texture1.c:202]: And here.

File: lib/texture2.c [hunk 15]
```diff
@@ -1505,7 +1494,7 @@ ktxTexture2_GetDataSizeUncompressed(ktxTexture2* This)
       case KTX_SS_BEGIN_VENDOR_RANGE:
       case KTX_SS_END_VENDOR_RANGE:
       case KTX_SS_BEGIN_RESERVED:
-	  default:
+          default:
```
Comment (MarkCallow) [inline lib/texture2.c:1497]: Wrong number of spaces.

File: libktx.gyp [hunk 16]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2015-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2015, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline libktx.gyp:6]: Remove this line.

File: tests/loadtests/common/vecmath.hpp [hunk 17]
```diff
@@ -15,30 +15,8 @@
...
- *
- * THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
- * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
- * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
- * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
- * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
- * MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+ * Copyright 2016-2020 Mark Callow
+ * SPDX-License-Identifier: MIT
```
Comment (MarkCallow) [inline tests/loadtests/common/vecmath.hpp:19]: Please change this to Apache 2.0.

File: tests/loadtests/glloadtests/shader-based/shaders.cpp [hunk 18]
```diff
@@ -2,38 +2,16 @@
...
-
-const GLchar* pszVs =
-"layout(location = 0) in vec4 position;\n"
-"layout(location = 1) in vec4 color;\n"
-"layout(location = 2) in vec4 texcoord;\n\n"
-
-"out vec4 v_color;\n"
-"out vec2 v_texcoord;\n\n"
+ * @author-2020 Mark Callow, www.edgewise-consulting.com
+ */ SPDX-License-Identifier: Apache-2.0
```
Comment (MarkCallow) [inline tests/loadtests/glloadtests/shader-based/shaders.cpp:14]: Delete these 2 lines? They look like  mistake. If they're meant to be here, an opening "/*" is needed.

File: tests/loadtests/vkloadtests/InstancedSampleBase.cpp [hunk 19]
```diff
@@ -164,9 +153,9 @@ InstancedSampleBase::InstancedSampleBase(VulkanContext& vkctx,
                 vk::FormatFeatureFlagBits::eSampledImageFilterLinear;
     }
     if (kTexture->generateMipmaps) {
-		neededFeatures |=  vk::FormatFeatureFlagBits::eBlitDst
-			    | vk::FormatFeatureFlagBits::eBlitSrc
-		       | vk::FormatFeatureFlagBits::eSampledImageFilterLinear;
+                neededFeatures |=  vk::FormatFeatureFlagBits::eBlitDst
+                            | vk::FormatFeatureFlagBits::eBlitSrc
+                       | vk::FormatFeatureFlagBits::eSampledImageFilterLinear;
```
Comment (MarkCallow) [inline tests/loadtests/vkloadtests/InstancedSampleBase.cpp:158]: Wrong tab size used.

Also, since this PR is removing most trailing spaces, lines 187, 190, 194 & 196 are blank lines with spaces on them. Please remove those spaces.

File: tests/testimages/genktx2 [hunk 20]
```diff
@@ -4,30 +4,8 @@
...
-# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-# MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
-# 
-# ---------------------------------------------------------------
+# Copyright 2017 The Khronos Group Inc.
+# SPDX-License-Identifier: MIT
```
Comment (MarkCallow) [inline tests/testimages/genktx2:8]: Change this to Apache 2.0.

File: tests/testimages/genref [hunk 21]
```diff
@@ -4,30 +4,8 @@
...
-# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
-# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-# MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
-# 
-# ---------------------------------------------------------------
+# Copyright 2017 The Khronos Group Inc.
+# SPDX-License-Identifier: MIT
```
Comment (MarkCallow) [inline tests/testimages/genref:8]: Change this one to Apache 2.0 too.

File: tests/transcodetests/transcodetests.cc [hunk 22]
```diff
@@ -113,13 +102,13 @@ bool isPo2(uint32_t i) {
...
+                return a + b;
 #if defined(_WIN32)
-	} else if (a.back() == UNIX_SEP) {
-		return a + b;
+        } else if (a.back() == UNIX_SEP) {
+                return a + b;
 #endif
-	} else {
+        } else {
         return a+OS_SEP+b;
```
Comment (MarkCallow) [inline tests/transcodetests/transcodetests.cc:112]: Wrong tab size in all the above.

File: tests/unittests/unittests.gypi [hunk 23]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2015-2020 Mark Callow.
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2015, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline tests/unittests/unittests.gypi:6]: Remove this line.

File: tools/toktx/toktx.gypi [hunk 24]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2015-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2015, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline tools/toktx/toktx.gypi:6]: Remove this line.

File: tools/tools.gypi [hunk 25]
```diff
@@ -1,3 +1,6 @@
+# Copyright 2019-2020 Mark Callow
+# SPDX-License-Identifier: Apache-2.0
+
 ##
 # @internal
 # @copyright © 2019, Mark Callow. For license see LICENSE.md.
```
Comment (MarkCallow) [inline tools/tools.gypi:6]: Remove this line.

**Comment by MarkCallow** [comment]
NOTICE.md is a symbolic link to LICENSE.md so I'm not sure how it could be out of date. However if you think it best to delete it then please do. You probably have a better understand of licensing terms that me.

File: .reuse/dep5 (hunk 3 above)
Comment (oddhack) [reply .reuse/dep5:114]: I was unsure because the tool referred to in the header of the file no longer seems to exist in that repo, but the naming and purpose of the repo was consistent.

File: LICENSES/GPL-2.0-or-later.txt (hunk 4 above)
Comment (oddhack) [reply LICENSES/GPL-2.0-or-later.txt:1]: It is one of the dual licenses on lib/zstddeclib.c.

Comment (oddhack) [reply LICENSES/GPL-2.0-or-later.txt:1]: dep5 is only for files that don't have copyright/license information imbedded in the file itself. For this repo that is binaries/images, or files that come from elsewhere I didn't feel comfortable modifying just for that purpose.

File: lib/texture1.c (hunk 12 above)
Comment (oddhack) [reply lib/texture1.c]: The various indentation things you noted should be fixed now.

File: tests/ktx2check-tests [hunk 26]
```diff
@@ -122,10 +111,10 @@ fi
...
 #  local tempfile=$(mktmp)
 #  for i in $*; do
-#	if [ ${i:0:2} == "--" ]; then
-#	  args="$args $i"
-#	  shift
-#	fi
+#       if [ ${i:0:2} == "--" ]; then
+#         args="$args $i"
+#         shift
+#       fi
```
[the remark this answers is not in the document]
Comment (oddhack) [reply tests/ktx2check-tests]: It is not an editor you've ever heard of before. The tabs were an accident, should be fixed now.

File: tests/loadtests/vkloadtests/InstancedSampleBase.cpp (hunk 19 above)
Comment (oddhack) [reply tests/loadtests/vkloadtests/InstancedSampleBase.cpp]: There are a lot of trailing blanks in the source files. It would be straightforward to run a script over all the C/C++ files looking for that, but not the purpose of this PR.

File: lib/basisu/CMakeLists.txt (hunk 10 above)
Comment (oddhack) [reply lib/basisu/CMakeLists.txt]: Done. I asked in https://github.com/BinomialLLC/basis_universal/issues/165
 about REUSE for their repo.

**Comment by MarkCallow** [comment]
> Oh, I also encourage importing new versions of the GL headers, though it's not clear to me if those are bound to the GLEW files also in the tree - if so that might be challenging.

I'll do that. Are they REUSE-compliant? khrplatform.h is not. Do you plan to update that?

File: tests/loadtests/glloadtests/shader-based/shaders.cpp [hunk 27]
```diff
@@ -2,39 +2,15 @@
...
-#include "ktx.h"
-
-const GLchar* pszVs =
-"layout(location = 0) in vec4 position;\n"
-"layout(location = 1) in vec4 color;\n"
-"layout(location = 2) in vec4 texcoord;\n\n"
-
-"out vec4 v_color;\n"
-"out vec2 v_texcoord;\n\n"
-
```
Comment (MarkCallow) [inline tests/loadtests/glloadtests/shader-based/shaders.cpp:37]: This block needs to be restored.

File: tests/loadtests/vkloadtests/InstancedSampleBase.cpp (hunk 19 above)
Comment (MarkCallow) [reply tests/loadtests/vkloadtests/InstancedSampleBase.cpp]: Do you have a script to do that? I'd love to get rid of the trailing blanks.

**Comment by MarkCallow** [comment]
I just merged PR #292 to pull in the updated dfdutils. Since there is no overlap, you shouldn't need to do anything on this PR because of it.

File: .reuse/dep5 (hunk 1 above)
Comment (oddhack) [reply .reuse/dep5:47]: I've dealt with all these except color_grid, which I assigned Apache 2.0 to for the moment.

**Comment by oddhack** [comment]
> I'll do that. Are they REUSE-compliant? khrplatform.h is not. Do you plan to update that?

Yes. There's a pending MR in the EGL repository it comes from.

**Comment by oddhack** [comment]
> I just merged PR #292 to pull in the updated dfdutils. Since there is no overlap, you shouldn't need to do anything on this PR because of it.

The files 'colourspaces.c' and 'queries.c' under lib/dfdutils/ don't appear to be present in the dfdutils repository, and don't have copyright/license information. Are these layered on top of the dfdutils subrepo somehow? There also seem to be some other differences between the files in lib/dfdutils in master branch here, and the current master in the dfdutils repo, although that doesn't affect REUSE.

File: tests/loadtests/vkloadtests/InstancedSampleBase.cpp (hunk 19 above)
Comment (oddhack) [reply tests/loadtests/vkloadtests/InstancedSampleBase.cpp]: Not per se, but something like

> find * -type f | egrep '\.(cc|cpp|hpp|c|h)$' | xargs sed -i -e 's/ *$//g'

would probably do the trick. Unless you have some source files with spaces in their names where a more complex find expression would be used instead of egrep to filter them out.

**Comment by oddhack** [comment]
> NOTICE.md is a symbolic link to LICENSE.md so I'm not sure how it could be out of date. However if you think it best to delete it then please do. You probably have a better understand of licensing terms that me.

I guess "incomplete" would be better than "out of date". While the list of exceptions could be brought up to speed, maintaining that information in two places doubles the pain. I'll take a look at the repo metafiles like LICENSE.md shortly, once the actual REUSE compliance is done.

**Comment by oddhack** [comment]
@MarkCallow this is getting close to being ready, but the MacOS build in CI is failing: https://travis-ci.com/github/KhronosGroup/KTX-Software/jobs/372375191 . The Linux build seems to be OK. Very likely I modified one of the Mac-specific files using the wrong comment convention, or something of that nature, since I am not familiar with the platform.

File: .reuse/dep5 [hunk 28]
```diff
@@ -10,12 +12,44 @@ Files: tests/srcimages/Iron_Bars/*
...
+License: CC0-1.0
+
+Files: tests/srcimages/FlightHelmet_baseColor.png
+Copyright: Microsoft
+License: CC0-1.0
+
+Files: tests/srcimages/CesiumLogoFlat_palette.png
+Copyright: Analytical Graphics
+License: CC-BY-4.0 WITH LicenseRef-Cesium-Trademark-Terms
+
```
Comment (MarkCallow) [inline .reuse/dep5:43]: Please add

```
Files: tests/testimages/himark.ktx tests/testimages/himark.ktx
Copyright: HI Corporation
License: LicenseRef-HI-Trademark
```

File: LICENSES/LicenseRef-Cesium-Trademark-Terms.txt [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+below.
+Laymen’s Terms: You need our permission to use our trademarks for any reasons not stated above. If you send us a
+request to use our trademarks, we can say no for any reason.
+
+Corporate Contact Info:
+Analytical Graphics, Inc.
+220 Valley Creek Blvd.
+Exton, Pennsylvania 19341
+1.610.981-8000
+contracts@agi.com
```
Comment (MarkCallow) [inline LICENSES/LicenseRef-Cesium-Trademark-Terms.txt:39]: Please add new file License-Ref/HI-Trademark

```
The HI logo textures are copyright by & trademarks of HI Corporation and are
provided for use only in testing the KTX loader. Any other use requires
specific prior written permission from HI. Furthermore the name HI may
not be used to endorse or promote products derived from this software
without specific prior written permission.
```

**Comment by MarkCallow** [comment]
Please make the following change to `ci_scripts/build_macos.sh` so we can see why the macos package build is failing.

```
@@ -50,7 +50,9 @@ ctest -C Release # --verbose
 echo "Install KTX-Software (macOS Release)"
 cmake --install . --config Release --prefix ../install-macos-release
 echo "Pack KTX-Software (macOS Release)"
-cpack -G productbuild
+if ! cpack -G productbuild; then
+  cat _CPack_Packages/Darwin/productbuild/ProductBuildOutput.log
+fi
 
 popd
```

File: cmake/License.rtf [hunk 30]
```diff
@@ -7,14 +7,14 @@
 \pard\pardeftab720\partightenfactor0
 
 \f0\fs26 \cf2 \expnd0\expndtw0\kerning0
-SPDX-License-Identifier: Apache-2.0\
+Copyright 2010-2020 The Khronos Group Inc.\
+SPDX-License-Identifier: Apache-2.0
```
Comment (MarkCallow) [inline cmake/License.rtf:11]: All text lines end in \\. I think the lack here is why the macos package build is failing.

**Comment by MarkCallow** [comment]
Re dfdutils, since this is where I use them this is where I develop new features and changes. I will push the updates upstream to KhronosGroup/dfdutils soon.

File: cmake/License.rtf (hunk 30 above)
Comment (oddhack) [reply cmake/License.rtf:11]: Not sure what to do about that. I directly edited the .rtf file because editing it in Libreoffice produced about 10x the volume of markup in the output, and I don't know RTF markup. Maybe you could edit it to insert the copyright/license using whatever editing tool you normally use, and attach it to the thread here?

**Comment by oddhack** [comment]
> Re dfdutils, since this is where I use them this is where I develop new features and changes. I will push the updates upstream to KhronosGroup/dfdutils soon.

So it sounds like I should make edits locally to the two new files that don't have licenses?

File: LICENSES/LicenseRef-Kodak.txt [hunk 31]
```diff
@@ -0,0 +1,7 @@
+This LICENSES file applies to images from the Kodak Lossless True Color
+Image Suite, and exists only for REUSE compliance. According to
+http://r0k.us/graphics/kodak/ , for such images:
```
[the remark this answers is not in the document]
Comment (oddhack) [reply LICENSES/LicenseRef-Kodak.txt:3]: I'm inclined to leave the white space in case someone tries to cut & paste the URL, though.

File: cmake/License.rtf (hunk 30 above)
Comment (oddhack) [reply cmake/License.rtf:11]: OK, just adding the trailing \ seems to work as far as libreoffice is concerned - NM.

**Comment by oddhack** [comment]
OK, I remembered the issue with License.rtf. Adding the trailing \, even with a space before it, confused the reuse tool to the point of throwing an error. I just ran the file through libreoffice and now it has the correct SPDX statement, passes reuse checking, and looks correct - it's just much bigger than previously.

**Comment by MarkCallow** [comment]
> So it sounds like I should make edits locally to the two new files that don't have licenses?

Yes and thanks for doing it.

Which of the 4 changes to `License.rtf`are necessary for`reuse`?
1. Removal of the blank line between SPDX & Copyright lines
2. Reordering the Copyright & SPDX lines
3. Changing the copyright symbol to the word.
4. Changing ` ~ ` between the copyright years to `-`.

Attached is a version with all 4 changes but without the double blank line after  the SPDX line and without the extra LibreOffice stuff. I had to zip it because GitHub doesn't support .rtf files.

[License.rtf.zip](https://github.com/KhronosGroup/KTX-Software/files/5081654/License.rtf.zip)

File: LICENSES/LicenseRef-Kodak.txt (hunk 31 above)
[the remark this answers is not in the document]
Comment (MarkCallow) [reply LICENSES/LicenseRef-Kodak.txt:3]: Okay.

File: .reuse/dep5 (hunk 1 above)
Comment (oddhack) [reply .reuse/dep5:51]: Maybe spin off an issue to track finding the original license on color_grid.png, but I don't think it's essential - if we get the license wrong the the creator can just ask us to change it.

**Comment by oddhack** [comment]
> Attached is a version with all 4 changes but without the double blank line after the SPDX line and without the extra LibreOffice stuff. I had to zip it because GitHub doesn't support .rtf files.

Thanks. Unfortunately, REUSE is still throwing an error on this file. I think its .rtf parsing is at fault but in any case, I did something different that worked around the problem by creating cmake/License.rtf.license, which is treated specially by the REUSE tool. Also filed an issue on their github repo to get the problem fixed.

With the latest commit, the repo passes REUSE and should be ready to merge whenever you choose. I did edit the LICENSE.md file to strip it down a lot, clarify the wide range of licenses in the repo, and refer to the reuse tool to construct a licensing bill of materials for the repo.

**Comment by oddhack** [comment]
@ntrevett I've attached a licensing bill of materials generated by 'reuse spdx', since that came up in the email discussion thread.
[bom.txt](https://github.com/KhronosGroup/KTX-Software/files/5083762/bom.txt)

**Comment by MarkCallow** [comment]
@oddhack how can I exclude some directories from `reuse`? It seems to be excluding a "build" directory already as it was successful on my local machine. However the CI build where I am running CI uses build-linux-debug and build-linux-release so I need to exclude those.

**Comment by oddhack** [comment]
TL;DR put the directories in .gitignore - see https://reuse.software/faq/#exclude-file

**Comment by MarkCallow** [comment]
Thanks Jon.
